src/button/buttonview.js

```diff
@@ -51,6 +53,25 @@ export default class ButtonView extends View {
 				}
 			}
 		};
+
+		this.register( 'children', el => el );
+	}
+
+	init() {
+		if ( this.model.icon ) {
+			this._setupIcon();
+		}
+
+		return super.init();
+	}
+
+	_setupIcon() {
+		const iconViewModel = new Model();
+		const iconView = new IconView( iconViewModel );
```

Comment:
`IconView.destroy()` could be called when `ButtonView` is destroyed. But I guess that we need some sort of utility to handle such a case.